.github/workflows/ci.yml

```diff
@@ -0,0 +1,42 @@
+name: Run Tests
+
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+
+jobs:
+  build_test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        perl:
+          - '5.38'
+          - '5.36'
+          - '5.34'
+          - '5.32'
+          - '5.30'
+          - '5.28'
+          - '5.26'
+          - '5.24'
+          - '5.22'
+    name: Perl ${{ matrix.perl }}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up perl
+        uses: shogo82148/actions-setup-perl@v1
+        with:
+          perl-version: ${{ matrix.perl }}
+
+      - run: perl -V
+
+      - name: Install Perl deps
+        run: |
+          cpanm --notest --installdeps .
+      - name: Run tests
+        run: |
```

Comment:
Note: I'd add
```
if: success()
```
to skip execution if dependencies are not installed

For repository with many tests I'd also add step to archive `.prove` in case of failure
so one can download it and use it to fix tests locally (maybe with usage of simple script [list-failed-tests](https://gist.github.com/happy-barney/a00bc41ff8f2b2e0ea4637479d1425fe))
```
   run: prove -lr --jobs $(nproc --all) --state save
name: Archive .prove (for 30 days)
   - uses: actions/upload-artifact@master
        if: failure()
        with:
          name: .prove
          path: .prove
```

Reply:
@happy-barney What happens with `if: success()` is used to skip tests if the `cpanm` fails. Does the workflow fail, too? (I assume so, but I'm not sure)

Reply:
yes, it does

Reply:
Added these suggestions. Thanks! Only difference is that I added `-${{ matrix.perl }}` as a suffix to the artifact name in case multiple jobs fail:

>           name: .prove-${{ matrix.perl }}

Reply:
Great, I didn't thought about that

Reply:
and oops, I made mistake in initial comment :-( should be expression
```
if: ${{ success() }}
```

for later (`.prove`) to run it only `id: run-test` step failed:
```
if: ${{ failure() && steps.run-tests.conclusion == 'failure' }}
```

Reply:
Updated. Thanks!